tf2_ros/src/tf2_ros/transform_listener.py

```diff
@@ -27,28 +27,29 @@
 
 # author: Wim Meeussen
 
-import roslib; roslib.load_manifest('tf2_ros')
-import rospy
+from rclpy.node import Node
 import tf2_ros
 from tf2_msgs.msg import TFMessage
 
-class TransformListener():
+class TransformListener(Node):
```

Comment:
`TransformListener`'s, `TransformBroadcaster`'s, and `StaticTransformBroadcaster`'s are often used as glorified subscribers/publishers inside a node with other (non-TF) subscribers/publishers. Why the added inheritance structure? 

Optionally creating a node (if none is supplied) under the hood makes sense to me, but I would usually call this class with a node that I'd prefer to re-use. Node re-use is essential if it [keeps the number of DDS Domain Participants down](https://community.rti.com/kb/what-maximum-number-participants-domain) on a given network.

I realize it's idiomatic in ROS2 to inherent from `Node`, but the use-case for passing in a pre-existing `Node` instance makes it a strong contender for an optional constructor argument.  This would also keep `tf2_ros` Python  more consistent with [`transform_listener`](https://github.com/ros2/geometry2/blob/fccfd986ad82208dc64d0f1d650dc87cf336cae7/tf2_ros/include/tf2_ros/transform_listener.h#L59), [`transform_broadcaster`](https://github.com/ros2/geometry2/blob/fccfd986ad82208dc64d0f1d650dc87cf336cae7/tf2_ros/include/tf2_ros/transform_broadcaster.h#L53) and [`static_transform_broadcaster`](https://github.com/ros2/geometry2/blob/fccfd986ad82208dc64d0f1d650dc87cf336cae7/tf2_ros/include/tf2_ros/static_transform_broadcaster.h#L55) in cpp land.

Reply:
Thanks for your comments. I guess a couple of months of me who was less experienced on ROS2 maybe wanted to follow the idiomatic in ROS2. I agree on the fact that the most important is keeping `tf2_ros` Python aligned with `tf2_ros`. I'll update PR.